File: mobile/lib/models/appointment.dart
class Appointment {
  final String id;
  final String clientId;
  final String garageId;
  final String serviceId;
  final DateTime date;
  final String time;
  final String status;
  final String? notes;
  final VehicleInfo? vehicleInfo;

  Appointment({
    required this.id,
    required this.clientId,
    required this.garageId,
    required this.serviceId,
    required this.date,
    required this.time,
    required this.status,
    this.notes,
    this.vehicleInfo,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['_id'] ?? json['id'] ?? '',
      clientId: json['clientId']?['_id'] ?? json['clientId'] ?? '',
      garageId: json['garageId']?['_id'] ?? json['garageId'] ?? '',
      serviceId: json['serviceId']?['_id'] ?? json['serviceId'] ?? '',
      date: DateTime.parse(json['date'] ?? DateTime.now().toIso8601String()),
      time: json['time'] ?? '',
      status: json['status'] ?? 'pending',
      notes: json['notes'],
      vehicleInfo: json['vehicleInfo'] != null
          ? VehicleInfo.fromJson(json['vehicleInfo'])
          : null,
    );
  }

  String get statusLabel {
    switch (status) {
      case 'pending':
        return 'En attente';
      case 'confirmed':
        return 'Confirmé';
      case 'in-progress':
        return 'En cours';
      case 'completed':
        return 'Terminé';
      case 'cancelled':
        return 'Annulé';
      default:
        return status;
    }
  }
}

class VehicleInfo {
  final String? brand;
  final String? model;
  final int? year;
  final String? licensePlate;

  VehicleInfo({
    this.brand,
    this.model,
    this.year,
    this.licensePlate,
  });

  factory VehicleInfo.fromJson(Map<String, dynamic> json) {
    return VehicleInfo(
      brand: json['brand'],
      model: json['model'],
      year: json['year'],
      licensePlate: json['licensePlate'],
    );
  }
}

File: backend/models/Appointment.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema(
  {
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    garageId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Garage',
      required: true,
    },
    serviceId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Service',
      required: true,
    },
    date: {
      type: Date,
      required: true,
    },
    time: {
      type: String,
      required: true,
    },
    status: {
      type: String,
      enum: ['pending', 'confirmed', 'in-progress', 'completed', 'cancelled'],
      default: 'pending',
    },
    paymentStatus: {
      type: String,
      enum: ['pending', 'paid', 'refunded', 'failed'],
      default: 'pending',
    },
    paymentIntentId: {
      type: String,
    },
    totalAmount: {
      type: Number,
      min: 0,
    },
    notes: {
      type: String,
      trim: true,
    },
    vehicleInfo: {
      brand: { type: String },
      model: { type: String },
      year: { type: Number },
      licensePlate: { type: String },
      photos: [{ type: String }], // URLs Cloudinary des photos du véhicule
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model('Appointment', appointmentSchema);

